hyperspy/_signals/hologram_image.py

```diff
@@ -0,0 +1,303 @@
+                          'sb_size': rec_param[4]}
+        wave_image.metadata.Signal.add_node('holo_rec_param')
+        wave_image.metadata.Signal.holo_rec_param.add_dictionary(rec_param_dict)
+        wave_image.axes_manager[0].scale = wave_image.axes_manager[0].scale*sx/wave.shape[0]
+        wave_image.axes_manager[1].scale = wave_image.axes_manager[1].scale*sy/wave.shape[1]
+        if show_phase:
+            phase = np.angle(wave)
+            f, ax = plt.subplots(1, 1)
+            ax.imshow(phase, cmap=cm.binary_r)
+            # f.canvas.manager.window.raise_()
+        if return_param:
+            return wave_image, rec_param
+        else:
+            return wave_image
+
+
+    @staticmethod
+    def _reconstruct(holo_data, sb_size, sb_pos, fresnel_ratio=0.3, fresnel_width=6):
+        """Core function for holographic reconstruction performing following steps:
+
+        * 2D FFT without apodisation;
+
+        * Cutting out sideband;
+
+        * Centering sideband;
+
+        * Applying round window;
+
+        * Applying sinc filter;
+
+        * Applying automatic filtering of Fresnel fringe (Fresnel filtering);
+
+        * Inverse FFT.
+
+        Parameters
+        ----------
+        holo_data : array_like
+            Holographic data array
+        sb_size : int
+            Size of the sideband filter in px
+        sb_pos : nparray (N=1)
+            Vector of two elements with sideband coordinates [y,x]
+        fresnel_ratio : float
+            The ratio of Fresnel filter with respect to the sideband size
+        fresnel_width : int
+            Width of fresnel fringe filter in px, set to 0 to disable filtering
+
+        Returns
+        -------
+            wav : nparray
+                Reconstructed electron wave
+
+        Notes
+        -----
+
+        Disabeling of Fresnel filter is not implemented
+
+        See Also
+        --------
+
+        reconstruct
+
+        """
+        # TODO: Parsing of the input has to be redone
+        # TODO: Add smoothing of Fresnel filter
+        # TODO: Add other smoothing options?
+
+        image_size = np.int(sb_size*2)
+        (sx, sy) = holo_data.shape
+        holo_data = np.float64(holo_data)
+
+        h_hw_fft = fftshift(fft2(holo_data))  # <---- NO Hanning filtering
+
+        sb_roi = h_hw_fft[(sb_pos[0]-image_size//2):(sb_pos[0]+image_size//2),
+                 (sb_pos[1]-image_size//2):(sb_pos[1]+image_size//2)]
+
+        (sb_ny, sb_nx) = sb_roi.shape
+        sb_l = min(sb_ny / 2, sb_nx / 2)
+        cen_yx = [sb_ny / 2, sb_nx / 2]
+
+        # Circular Aperture
+        cgrid_y = np.arange(-sb_ny / 2, sb_ny / 2, 1)
+        cgrid_x = np.arange(-sb_nx / 2, sb_nx / 2, 1)
+        (sb_xx, sb_yy) = np.meshgrid(cgrid_x, cgrid_y)
+        sb_r = np.sqrt(sb_xx ** 2 + sb_yy ** 2)
+        c_mask = np.zeros((sb_nx, sb_ny))  # Original: cMask=zeros(SB_Ny,SB_Nx);
+
+        c_mask[sb_r < sb_size] = 1
```

Comment:
All instances of `sb_nx` and `sb_ny` can be replaced by `image_size` as shown above.
Also you can use a slightly more elegant solution to replace the first three lines:
```python
sb_yy, sb_xx = np.indices((image_size, image_size)) - sb_size
```